src/poolboy.erl

```diff
@@ -280,37 +277,41 @@ handle_checkin(Pid, State) ->
            waiting = Waiting,
            monitors = Monitors,
            overflow = Overflow} = State,
-    case queue:out(Waiting) of
-        {{value, {From, Ref}}, Left} ->
-            true = ets:insert(Monitors, {Pid, Ref}),
-            gen_server:reply(From, Pid),
-            State#state{waiting = Left};
-        {empty, Empty} when Overflow > 0 ->
+    case ets:first(Waiting) of
```

Comment:
The `Waiting` ets table is of type set so dequeues are random or rather hash based. This is interesting but likely leads to the large 99% of LIFO queues because a process hashed to an unlucky bucket will effectively be at the end of the queue, with newer process jumping ahead of it.